diff -ruN bzip2-1.0.6.patched/Makefile-libbz2_so bzip2-1.0.6/Makefile-libbz2_so
--- bzip2-1.0.6/Makefile-libbz2_so	2019-04-03 06:44:24.552005001 +0000
+++ bzip2-1.0.6.patched/Makefile-libbz2_so	2019-04-03 06:50:35.932005001 +0000
@@ -35,8 +35,8 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
+	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS) $(LDFLAGS)
+	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6 $(LDFLAGS)
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.6 libbz2.so.1.0
 
