--- trunk/gcc/testsuite/gcc.dg/20020312-2.c	2018/09/25 07:30:56	264564
+++ trunk/gcc/testsuite/gcc.dg/20020312-2.c	2018/09/25 12:25:44	264565
@@ -7,6 +7,7 @@
 
 /* { dg-do run } */
 /* { dg-options "-O -fno-pic" } */
+/* { dg-additional-options "-no-pie" { target pie_enabled } } */
 /* { dg-require-effective-target nonlocal_goto } */
 
 extern void abort (void);
--- trunk/gcc/testsuite/gcc.dg/uninit-19.c	2018/09/25 07:30:56	264564
+++ trunk/gcc/testsuite/gcc.dg/uninit-19.c	2018/09/25 12:25:44	264565
@@ -12,7 +12,7 @@
 {
   if (p1 & 8)
     b[3] = p10[a];
-  /* { dg-warning "may be used uninitialized" "" { target { { nonpic } || { hppa*64*-*-* } } } .-1 } */
+  /* { dg-warning "may be used uninitialized" "" { target { { nonpic || pie_enabled } || { hppa*64*-*-* } } } .-1 } */
 }
 
 void
@@ -22,5 +22,5 @@
   if (l & 6)
     n = &c + m;
   fn1 (l, &d, &e, &g, &i, &h, &k, n);
-  /* { dg-warning "may be used uninitialized" "" { target { ! { { nonpic } || { hppa*64*-*-* } } } } .-1 } */
+  /* { dg-warning "may be used uninitialized" "" { target { ! { { nonpic || pie_enabled } || { hppa*64*-*-* } } } } .-1 } */
 }
--- trunk/gcc/testsuite/gcc.target/i386/pr34256.c	2018/09/25 07:30:56	264564
+++ trunk/gcc/testsuite/gcc.target/i386/pr34256.c	2018/09/25 12:25:44	264565
@@ -10,5 +10,5 @@
   return _mm_cvtm64_si64(_mm_add_pi32(x, y));
 }
 
-/* { dg-final { scan-assembler-times "mov" 2 { target nonpic } } } */
-/* { dg-final { scan-assembler-times "mov" 4 { target { ! nonpic } } } } */
+/* { dg-final { scan-assembler-times "mov" 2 { target { nonpic || pie_enabled } } } } */
+/* { dg-final { scan-assembler-times "mov" 4 { target { { ! nonpic } && { ! pie_enabled } } } } } */
