diff -ruN gcc-8.2.0/gcc/testsuite/g++.dg/asan/asan_test.C gcc-8.2.0.patched/gcc/testsuite/g++.dg/asan/asan_test.C
--- gcc-8.2.0/gcc/testsuite/g++.dg/asan/asan_test.C	2017-10-19 20:23:59.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/g++.dg/asan/asan_test.C	2019-03-30 21:43:59.601057897 +0900
@@ -2,7 +2,7 @@
 // { dg-skip-if "" { *-*-* } { "*" } { "-O2" } }
 // { dg-skip-if "" { *-*-* } { "-flto" } { "" } }
 // { dg-additional-sources "asan_globals_test-wrapper.cc" }
-// { dg-options "-std=c++11 -fsanitize=address -fno-builtin -Wall -Werror -g -DASAN_UAR=0 -DASAN_HAS_EXCEPTIONS=1 -DASAN_HAS_BLACKLIST=0 -DSANITIZER_USE_DEJAGNU_GTEST=1 -lasan -lpthread -ldl" }
+// { dg-options "-std=c++11 -fsanitize=address -fno-builtin -Wall -Werror -g -DASAN_UAR=0 -DASAN_HAS_EXCEPTIONS=1 -DASAN_HAS_BLACKLIST=0 -DSANITIZER_USE_DEJAGNU_GTEST=1 -lasan -lpthread -ldl -fno-stack-protector" }
 // { dg-additional-options "-DASAN_NEEDS_SEGV=1" { target { ! arm*-*-* } } }
 // { dg-additional-options "-DASAN_LOW_MEMORY=1 -DASAN_NEEDS_SEGV=0" { target arm*-*-* } }
 // { dg-additional-options "-DASAN_AVOID_EXPENSIVE_TESTS=1" { target { ! run_expensive_tests } } }
diff -ruN gcc-8.2.0/gcc/testsuite/g++.dg/ubsan/vla-1.C gcc-8.2.0.patched/gcc/testsuite/g++.dg/ubsan/vla-1.C
--- gcc-8.2.0/gcc/testsuite/g++.dg/ubsan/vla-1.C	2016-04-15 02:35:23.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/g++.dg/ubsan/vla-1.C	2019-03-30 21:44:03.585057734 +0900
@@ -1,5 +1,5 @@
 // { dg-do run }
-// { dg-options "-Wno-vla -fsanitize=undefined" }
+// { dg-options "-Wno-vla -fsanitize=undefined -fno-stack-protector" }
 // { dg-output "index 1 out of bounds" }
 
 void f(int i) {
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.dg/asan/use-after-scope-4.c gcc-8.2.0.patched/gcc/testsuite/gcc.dg/asan/use-after-scope-4.c
--- gcc-8.2.0/gcc/testsuite/gcc.dg/asan/use-after-scope-4.c	2017-06-13 22:59:05.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.dg/asan/use-after-scope-4.c	2019-03-30 21:42:13.337062246 +0900
@@ -1,4 +1,5 @@
 // { dg-do run }
+// { dg-options "-fno-stack-protector" }
 
 #define FN(NAME) \
 NAME (void) \
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.dg/stack-usage-1.c gcc-8.2.0.patched/gcc/testsuite/gcc.dg/stack-usage-1.c
--- gcc-8.2.0/gcc/testsuite/gcc.dg/stack-usage-1.c	2017-06-18 00:32:28.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.dg/stack-usage-1.c	2019-03-30 21:42:26.993061687 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile } */
-/* { dg-options "-fstack-usage" } */
+/* { dg-options "-fstack-usage -fno-stack-protector" } */
 /* nvptx doesn't have a reg allocator, and hence no stack usage data.  */
 /* { dg-skip-if "" { nvptx-*-* } } */
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.dg/superblock.c gcc-8.2.0.patched/gcc/testsuite/gcc.dg/superblock.c
--- gcc-8.2.0/gcc/testsuite/gcc.dg/superblock.c	2015-05-29 17:20:29.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.dg/superblock.c	2019-03-30 21:42:39.489061176 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile } */
-/* { dg-options "-O2 -fno-asynchronous-unwind-tables -fsched2-use-superblocks -fdump-rtl-sched2 -fdump-rtl-bbro" } */
+/* { dg-options "-O2 -fno-asynchronous-unwind-tables -fsched2-use-superblocks -fdump-rtl-sched2 -fdump-rtl-bbro -fno-stack-protector" } */
 /* { dg-require-effective-target scheduling } */
 
 typedef int aligned __attribute__ ((aligned (64)));
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/avx-vzeroupper-17.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/avx-vzeroupper-17.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/avx-vzeroupper-17.c	2014-09-22 00:13:14.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/avx-vzeroupper-17.c	2019-03-30 21:42:45.729060920 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target lp64 } } */
-/* { dg-options "-O2 -mavx -mabi=ms -dp" } */
+/* { dg-options "-O2 -mavx -mabi=ms -dp -fno-stack-protector" } */
 
 typedef float __m256 __attribute__ ((__vector_size__ (32), __may_alias__));
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/interrupt-redzone-1.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/interrupt-redzone-1.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/interrupt-redzone-1.c	2016-06-04 00:08:00.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/interrupt-redzone-1.c	2019-03-30 21:43:17.997059600 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { ! ia32 } } } */
-/* { dg-options "-O2 -mgeneral-regs-only -mno-cld -mred-zone" } */
+/* { dg-options "-O2 -mgeneral-regs-only -mno-cld -mred-zone -fno-stack-protector" } */
 
 void
 __attribute__((interrupt))
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/interrupt-redzone-2.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/interrupt-redzone-2.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/interrupt-redzone-2.c	2016-06-10 03:31:35.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/interrupt-redzone-2.c	2019-03-30 21:43:21.401059460 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { ! ia32 } } } */
-/* { dg-options "-O2 -mgeneral-regs-only -mno-cld -mred-zone" } */
+/* { dg-options "-O2 -mgeneral-regs-only -mno-cld -mred-zone -fno-stack-protector" } */
 
 void
 __attribute__((interrupt))
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-1.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-1.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-1.c	2015-08-28 03:29:37.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-1.c	2019-03-30 21:42:52.713060635 +0900
@@ -1,6 +1,6 @@
 /* PR target/67317 */
 /* { dg-do compile } */
-/* { dg-options "-O2" } */
+/* { dg-options "-O2 -fno-stack-protector" } */
 
 typedef unsigned int u32;
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-2.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-2.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-2.c	2015-08-28 03:29:37.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-2.c	2019-03-30 21:42:57.977060419 +0900
@@ -1,6 +1,6 @@
 /* PR target/67317 */
 /* { dg-do compile { target { ! ia32 } } } */
-/* { dg-options "-O2" } */
+/* { dg-options "-O2 -fno-stack-protector" } */
 
 typedef unsigned long long u64;
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-3.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-3.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-3.c	2015-08-28 03:29:37.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-3.c	2019-03-30 21:43:03.073060211 +0900
@@ -1,6 +1,6 @@
 /* PR target/67317 */
 /* { dg-do compile } */
-/* { dg-options "-O2" } */
+/* { dg-options "-O2 -fno-stack-protector" } */
 
 typedef unsigned int u32;
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-4.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-4.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr67317-4.c	2015-08-28 03:29:37.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr67317-4.c	2019-03-30 21:43:06.593060067 +0900
@@ -1,6 +1,6 @@
 /* PR target/67317 */
 /* { dg-do compile { target { ! ia32 } } } */
-/* { dg-options "-O2" } */
+/* { dg-options "-O2 -fno-stack-protector" } */
 
 typedef unsigned long long u64;
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr68961.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr68961.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr68961.c	2016-07-12 17:56:14.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr68961.c	2019-03-30 21:43:24.193059346 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target lp64 } } */
-/* { dg-options "-O3 -fno-vect-cost-model -fdump-tree-slp2-details" } */
+/* { dg-options "-O3 -fno-vect-cost-model -fdump-tree-slp2-details -fno-stack-protector" } */
 
 struct x { double d[2]; };
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr79793-1.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr79793-1.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr79793-1.c	2017-07-31 23:59:24.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr79793-1.c	2019-03-30 21:43:27.721059202 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { *-*-linux* && { ! ia32 } } } } */
-/* { dg-options "-O2 -mgeneral-regs-only -mtune=generic" } */
+/* { dg-options "-O2 -mgeneral-regs-only -mtune=generic -fno-stack-protector" } */
 
 void
  __attribute__ ((interrupt))
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr79793-2.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr79793-2.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr79793-2.c	2017-07-31 23:59:24.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr79793-2.c	2019-03-30 21:43:30.593059084 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { *-*-linux* && { ! ia32 } } } } */
-/* { dg-options "-O2 -mgeneral-regs-only -mtune=generic" } */
+/* { dg-options "-O2 -mgeneral-regs-only -mtune=generic -fno-stack-protector" } */
 
 typedef unsigned int uword_t __attribute__ ((mode (__word__)));
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/shrink_wrap_1.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/shrink_wrap_1.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/shrink_wrap_1.c	2015-05-29 17:20:29.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/shrink_wrap_1.c	2019-03-30 21:43:09.649059941 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { ! ia32 } } } */
-/* { dg-options "-O2 -fdump-rtl-pro_and_epilogue" } */
+/* { dg-options "-O2 -fdump-rtl-pro_and_epilogue -fno-stack-protector" } */
 
 enum machine_mode
 {
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/stack-check-11.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/stack-check-11.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/stack-check-11.c	2017-10-04 02:09:54.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/stack-check-11.c	2019-03-30 21:43:34.401058928 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile } */
-/* { dg-options "-O2 -fstack-clash-protection -mtune=generic" } */
+/* { dg-options "-O2 -fstack-clash-protection -mtune=generic -fno-stack-protector" } */
 /* { dg-require-effective-target supports_stack_clash_protection } */
 
 #include <stdint.h>
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/stack-check-18.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/stack-check-18.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/stack-check-18.c	2018-04-21 02:20:46.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/stack-check-18.c	2019-03-30 21:43:48.657058345 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile } */
-/* { dg-options "-O2 -fstack-clash-protection -mtune=generic -fdump-rtl-expand" } */
+/* { dg-options "-O2 -fstack-clash-protection -mtune=generic -fdump-rtl-expand -fno-stack-protector" } */
 /* { dg-require-effective-target supports_stack_clash_protection } */
 /* { dg-skip-if "" { *-*-* } { "-fstack-protector*" } { "" } } */
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/stack-check-19.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/stack-check-19.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/stack-check-19.c	2018-04-21 02:20:46.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/stack-check-19.c	2019-03-30 21:43:51.529058227 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile } */
-/* { dg-options "-O2 -fstack-clash-protection -mtune=generic -fdump-rtl-expand" } */
+/* { dg-options "-O2 -fstack-clash-protection -mtune=generic -fdump-rtl-expand -fno-stack-protector" } */
 /* { dg-require-effective-target supports_stack_clash_protection } */
 /* { dg-skip-if "" { *-*-* } { "-fstack-protector*" } { "" } } */
 
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/sw-1.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/sw-1.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/sw-1.c	2017-06-18 00:32:28.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/sw-1.c	2019-03-30 21:43:14.409059747 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile } */
-/* { dg-options "-O2 -mtune=generic -fshrink-wrap -fdump-rtl-pro_and_epilogue" } */
+/* { dg-options "-O2 -mtune=generic -fshrink-wrap -fdump-rtl-pro_and_epilogue -fno-stack-protector" } */
 /* { dg-skip-if "No shrink-wrapping preformed" { x86_64-*-mingw* } } */
 
 #include <string.h>
