diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/cet-sjlj-6a.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/cet-sjlj-6a.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/cet-sjlj-6a.c	2018-04-25 01:54:35.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/cet-sjlj-6a.c	2019-03-30 21:45:59.497052990 +0900
@@ -1,6 +1,6 @@
 /* { dg-do compile { target { ! ia32 } } } */
 /* { dg-require-effective-target maybe_x32 } */
-/* { dg-options "-O -maddress-mode=short -fcf-protection -mx32" } */
+/* { dg-options "-fno-PIE -no-pie -O -maddress-mode=short -fcf-protection -mx32" } */
 /* { dg-final { scan-assembler-times "endbr64" 2 } } */
 /* { dg-final { scan-assembler-times "movq\t.*buf\\+8" 1 } } */
 /* { dg-final { scan-assembler-times "subq\tbuf\\+8" 1 } } */
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/cet-sjlj-6b.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/cet-sjlj-6b.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/cet-sjlj-6b.c	2018-04-25 01:54:35.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/cet-sjlj-6b.c	2019-03-30 21:46:13.881052401 +0900
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { ! ia32 } } } */
-/* { dg-options "-O -maddress-mode=long -fcf-protection -mx32" } */
+/* { dg-options "-fno-PIE -no-pie -O -maddress-mode=long -fcf-protection -mx32" } */
 /* { dg-final { scan-assembler-times "endbr64" 2 } } */
 /* { dg-final { scan-assembler-times "movq\t.*buf\\+16" 1 } } */
 /* { dg-final { scan-assembler-times "subq\tbuf\\+16" 1 } } */
diff -ruN gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr24414.c gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr24414.c
--- gcc-8.2.0/gcc/testsuite/gcc.target/i386/pr24414.c	2016-06-06 21:31:59.000000000 +0900
+++ gcc-8.2.0.patched/gcc/testsuite/gcc.target/i386/pr24414.c	2019-03-30 21:46:20.793052118 +0900
@@ -1,5 +1,5 @@
 /* { dg-do run } */
-/* { dg-options "-O2" } */
+/* { dg-options "-fno-PIE -no-pie -O2" } */
 int test;
 
 int
